servicetalk-data-protobuf/src/test/java/io/servicetalk/data/protobuf/ProtobufSerializerFactoryTest.java

```diff
@@ -143,4 +149,114 @@ private static DummyMessage newMsg(int length) {
         }
         return newMsg(sb.toString());
     }
+
+    @Test
+    void testProtobufV2Compatibility() {
+        SerializerDeserializer<Protobuf2Message> serializerDeserializer =
+                ProtobufSerializerFactory.PROTOBUF.serializerDeserializer(Protobuf2Message.class);
+
+        Buffer dummyBuffer = DEFAULT_ALLOCATOR.newBuffer();
+        Protobuf2Message deserialized = serializerDeserializer.deserialize(dummyBuffer, DEFAULT_ALLOCATOR);
+        assertThat(deserialized, sameInstance(Protobuf2Message.MESSAGE));
+    }
+
+    @SuppressWarnings("PMD.MutableStaticState")
+    public static class Protobuf2Message extends AbstractMessageLite<Protobuf2Message, Protobuf2Message.Builder> {
+
+        static final Protobuf2Message MESSAGE = new Protobuf2Message() {
+            @Override
+            public boolean isInitialized() {
+                return true;
+            }
+        };
+
+        /**
+         * This declaration is intended to be compatible with how protoc 2.5 generated message classes.
```

Comment:
to test compatibility we should create a new project that brings in protobuf 2.x as a test dependency, generates some code, and verify the serializer works. there maybe other things that we change/depend upon over time. I also don't think we want to commit/guarantee protnbuf 2 compatibility externally ([protobuf 3 was released in 2016](https://github.com/protocolbuffers/protobuf/releases/tag/v3.0.0), last [protobuf 2 release was in 2014](https://github.com/protocolbuffers/protobuf/releases/tag/v2.6.1)), and if we have to advertise anything it would be "best effort" as this may restrict our ability to move forward with new features, etc.